CefSharp.Wpf.Example/Views/BrowserTabView.xaml.cs

```diff
@@ -85,6 +89,26 @@ public BrowserTabView()
             CefExample.RegisterTestResources(browser);
         }
 
+        private void OnBeforeDownloadFired(object sender, DownloadItem e)
+        {
+            this.UpdateDownloadAction("OnBeforeDownload", e);
+        }
+
+        private void OnDownloadUpdatedFired(object sender, DownloadItem e)
+        {
+            this.UpdateDownloadAction("OnDownloadUpdated", e);
+        }
+
+        private void UpdateDownloadAction(string downloadAction, DownloadItem downloadItem)
+        {
+            this.Dispatcher.Invoke(() =>
```

Comment:
Is there a reason to update in a `sync` fashion? `UI` updates should be `async` where possible.